tests/unit/composables/PluginManager.spec.js

```diff
@@ -258,22 +258,42 @@ describe('Test composable: PluginManager', () => {
   });
 
   describe('Test function: addNewComponent', () => {
-    it('should call addComponent and render', async () => {
-      const plugin = {
-        data: {
-          addComponent: jest.fn(),
-        },
-        isParsable: () => true,
-        render: jest.fn(() => []),
-      };
-
+    const plugin = {
+      data: {
+        addComponent: jest.fn(),
+        getComponentById: jest.fn(),
+      },
+      isParsable: () => true,
+      render: jest.fn(() => []),
+    };
+
+    plugin.data.getComponentById.mockReturnValue({ drawOption: null });
+
+    it('should call addComponent and render without position', async () => {
       expect(plugin.render).toHaveBeenCalledTimes(0);
 
       await PluginManager.addNewComponent('projectName', plugin, 'plugin/model', {});
 
       expect(plugin.data.addComponent).toHaveBeenCalledTimes(1);
+      expect(plugin.data.getComponentById).toHaveBeenCalledTimes(0);
       expect(plugin.render).toHaveBeenCalledTimes(1);
     });
+
+    it('should call addComponent and render with position', async () => {
+      expect(plugin.render).toHaveBeenCalledTimes(1);
+
+      await PluginManager.addNewComponent(
+        'projectName',
+        plugin,
+        'plugin/model',
+        {},
+        { x: 0, y: 0 },
+      );
+
+      expect(plugin.data.addComponent).toHaveBeenCalledTimes(2);
+      expect(plugin.data.getComponentById).toHaveBeenCalledTimes(1);
+      expect(plugin.render).toHaveBeenCalledTimes(2);
```

Comment:
Unless there is a reason for `render` and `addComponent` to be called twice, try to mock these functions so they are only called once

Reply:
Done